modules/minimp3/audio_stream_mp3.cpp

```diff
@@ -159,7 +159,13 @@ void AudioStreamMP3::set_data(const Vector<uint8_t> &p_data) {
 	const uint8_t *src_datar = p_data.ptr();
 
 	mp3dec_ex_t mp3d;
-	mp3dec_ex_open_buf(&mp3d, src_datar, src_data_len, MP3D_SEEK_TO_SAMPLE);
+	int err = mp3dec_ex_open_buf(&mp3d, src_datar, src_data_len, MP3D_SEEK_TO_SAMPLE);
+	if (err == MP3D_E_PARAM) {
+		WARN_PRINT("mp3dec_ex_open_buf returned error code MP3D_E_PARAM");
+		return;
```

Comment:
This should be an error and not a warning IMO, as it effectively aborts the operation.
So that would typically be:
```
ERR_FAIL_COND(err == MP3D_E_PARAM);
```

which will generate an error like:
```
set_data: Condition "err == MP3D_E_PARAM" is true. Breaking.
At:   modules/minimp3/audio_stream_mp3.cpp:163.
```

Or `ERR_FAIL_COND_MSG` if you want to give a custom message, but IMO unless we explain what `MP3D_E_PARAM` means, the default message is good enough (lets you know where to put a breakpoint if you want to debug what's wrong).

Reply:
I'd make it error out on all minimp3 errors though, looking at the list they all seem critical apart from `MP3D_E_USER`, but that one has specifically handling to *not* error out in `mp3dec_ex_open_buf`:
```
/* return error codes */
#define MP3D_E_PARAM   -1
#define MP3D_E_MEMORY  -2
#define MP3D_E_IOERROR -3
#define MP3D_E_USER    -4  /* can be used to stop processing from callbacks without indicating specific error */
#define MP3D_E_DECODE  -5  /* decode error which can't be safely skipped, such as sample rate, layer and channels change */
```